deepspeed/runtime/engine.py

```diff
@@ -234,6 +233,7 @@ def __init__(
 
         global dist
         from deepspeed import comm as dist
```

Comment:
may be just moving this import to the top of the file and then `global dist` is no longer needed? I could be wrong

I did a quick test with the following and it seems to work:
```
diff --git a/deepspeed/runtime/engine.py b/deepspeed/runtime/engine.py
index f3480f76..e9541092 100644
--- a/deepspeed/runtime/engine.py
+++ b/deepspeed/runtime/engine.py
@@ -95,6 +95,7 @@ from deepspeed.accelerator import get_accelerator
 from deepspeed.ops.op_builder import UtilsBuilder

 from deepspeed.runtime.config import DtypeEnum
+from deepspeed import comm as dist

 MEMORY_OPT_ALLREDUCE_SIZE = 500000000

@@ -141,8 +142,6 @@ BACKWARD_REDUCE_GLOBAL_TIMER = 'backward_allreduce'
 STEP_MICRO_TIMER = 'step_microstep'
 STEP_GLOBAL_TIMER = 'step'

-dist = None
-

 class EngineTimers(object):
     r"""Wallclock timers for DeepSpeedEngine"""
@@ -231,9 +230,6 @@ class DeepSpeedEngine(Module):

         self.checkpoint_engine = None

-        global dist
-        from deepspeed import comm as dist
-
         self._is_gradient_accumulation_boundary = None
         self.scale_wrt_gas = None

```